astropy/cosmology/_src/utils.py

```diff
@@ -68,32 +68,51 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+# ===================================================================
+
+ScalarTypes: TypeAlias = Number | np.generic
+SCALAR_TYPES: Final = (float, int, np.generic, Number)  # arranged for speed
+
+
 class HasShape(Protocol):
     shape: tuple[int, ...]
 
 
 def aszarr(
-    z: Number | np.generic | HasShape | ArrayLike,
-) -> Number | np.generic | NDArray[Any]:
-    """Redshift as a `~numbers.Number` or |ndarray| / |Quantity| / |Column|.
+    z: Quantity | NDArray[Any] | ArrayLike | ScalarTypes | HasShape, /
+) -> NDArray[Any]:
+    """Redshift as an Array duck type.
 
     Allows for any ndarray ducktype by checking for attribute "shape".
     """
-    if isinstance(z, (Number, np.generic)):  # scalars
+    # Scalars
+    if isinstance(z, SCALAR_TYPES):
+        return np.asarray(z)
+
+    # Arrays
+    # Need to special-case Quantity since it is an NDArray subclass
```

Comment:
So from previous discussion, I'd replace from here to end with,
```
if isinstance(z, np.ndarray) and not isinstance(z, Quantity):
    return z

return Quantity(z, cu.redshift, copy=False, subok=True).view(np.ndarray)  # or .value[...], but it's slower.
```

Reply:
Done, thanks!

Reply:
This is proving problematic. `copy=False` isn't always possible. `copy=None` isn't supported in the oldest version of numpy.